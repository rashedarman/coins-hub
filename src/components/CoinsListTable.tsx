import { Table, Text } from '@mantine/core';
import { useNavigate } from 'react-router-dom';
import { Coin } from '../interfaces';
import { formatCoinPrice, priceToKMB } from '../utils';
import CoinIconNameSymbol from './CoinNameIcon';
import PriceChangeBadge from './PriceChangeBadge';

interface Props {
  coins: Coin[];
}

const CoinsListTable = (props: Props) => {
  const { coins } = props;

  const navigate = useNavigate();

  return (
    <Table
      highlightOnHover
      withBorder
      verticalSpacing="lg"
      horizontalSpacing="xl"
      my="xl"
    >
      <thead>
        <tr>
          <th>#</th>
          <th>Name</th>
          <th>Change (24h)</th>
          <th>Price</th>
          <th>Market Cap</th>
          <th>Volume 24h</th>
        </tr>
      </thead>
      <tbody>
        {coins.map((coin) => (
          <tr
            key={coin.id}
            onClick={() => navigate(`/coins/${coin.id}`)}
            style={{ cursor: 'pointer' }}
          >
            <td>{coin.rank}</td>
            <td>
              <CoinIconNameSymbol
                icon={coin.icon}
                name={coin.name}
                symbol={coin.symbol}
              />
            </td>
            <td>
              <PriceChangeBadge priceChange={coin.priceChange1d} />
            </td>
            <td>
              <Text fw="500">{formatCoinPrice(coin.price)}</Text>
            </td>
            <td>
              <Text>
                $
                {priceToKMB(coin.marketCap)}
              </Text>
            </td>
            <td>
              <Text>
                $
                {priceToKMB(coin.volume)}
              </Text>
            </td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default CoinsListTable;
