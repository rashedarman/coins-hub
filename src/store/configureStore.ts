import { configureStore } from '@reduxjs/toolkit';
import cryptoSlice from './crypto/cryptoSlice';
import newsSlice from './news/newsSlice';

const rootReducer = {
  crypto: cryptoSlice,
  news: newsSlice,
};

const store = configureStore({
  reducer: rootReducer,
});

export default store;
