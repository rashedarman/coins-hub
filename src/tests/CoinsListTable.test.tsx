import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CoinsListTable from '../components/CoinsListTable';
import store from '../store/configureStore';
import { coins } from './__mockData';

describe('Testing CoinsListTable component', () => {
  it('Renders CoinsListTable component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CoinsListTable coins={coins} />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
