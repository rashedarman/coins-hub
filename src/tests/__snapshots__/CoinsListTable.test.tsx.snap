// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CoinsListTable component Renders CoinsListTable component correctly 1`] = `
<div>
  <table
    class="mantine-Table-root mantine-udihb2"
    data-hover="true"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Name
        </th>
        <th>
          Change (24h)
        </th>
        <th>
          Price
        </th>
        <th>
          Market Cap
        </th>
        <th>
          Volume 24h
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        style="cursor: pointer;"
      >
        <td>
          1
        </td>
        <td>
          <div
            class="mantine-xg7kom"
          >
            <div
              class="mantine-Image-root mantine-wbcrf9"
              style="width: 2rem;"
            >
              <figure
                class="mantine-11nhzn5 mantine-Image-figure"
              >
                <div
                  class="mantine-qqmv3w mantine-Image-imageWrapper"
                >
                  <img
                    alt="Bitcoin"
                    class="mantine-3y8yz3 mantine-Image-image"
                    src="https://static.coinstats.app/coins/1650455588819.png"
                    style="object-fit: cover; width: 2rem; height: auto;"
                  />
                </div>
              </figure>
            </div>
            <div
              class="mantine-Text-root mantine-qdrdlf"
            >
              Bitcoin
            </div>
            <div
              class="mantine-Text-root mantine-re18hr"
            >
              BTC
            </div>
          </div>
        </td>
        <td>
          <div>
            <div
              class="mantine-Badge-root mantine-gsj07c"
            >
              <span
                class="mantine-h9iq4m mantine-Badge-inner"
              >
                ▼
                -0.15
                %
              </span>
            </div>
          </div>
        </td>
        <td>
          <div
            class="mantine-Text-root mantine-1bk40a"
          >
            $22,959.00
          </div>
        </td>
        <td>
          <div
            class="mantine-Text-root mantine-1pbxw0k"
          >
            $
            442.8B
          </div>
        </td>
        <td>
          <div
            class="mantine-Text-root mantine-1pbxw0k"
          >
            $
            36.4B
          </div>
        </td>
      </tr>
      <tr
        style="cursor: pointer;"
      >
        <td>
          2
        </td>
        <td>
          <div
            class="mantine-xg7kom"
          >
            <div
              class="mantine-Image-root mantine-wbcrf9"
              style="width: 2rem;"
            >
              <figure
                class="mantine-11nhzn5 mantine-Image-figure"
              >
                <div
                  class="mantine-qqmv3w mantine-Image-imageWrapper"
                >
                  <img
                    alt="Ethereum"
                    class="mantine-3y8yz3 mantine-Image-image"
                    src="https://static.coinstats.app/coins/1633034449328.png"
                    style="object-fit: cover; width: 2rem; height: auto;"
                  />
                </div>
              </figure>
            </div>
            <div
              class="mantine-Text-root mantine-qdrdlf"
            >
              Ethereum
            </div>
            <div
              class="mantine-Text-root mantine-re18hr"
            >
              ETH
            </div>
          </div>
        </td>
        <td>
          <div>
            <div
              class="mantine-Badge-root mantine-gsj07c"
            >
              <span
                class="mantine-h9iq4m mantine-Badge-inner"
              >
                ▼
                -0.24
                %
              </span>
            </div>
          </div>
        </td>
        <td>
          <div
            class="mantine-Text-root mantine-1bk40a"
          >
            $1,727.00
          </div>
        </td>
        <td>
          <div
            class="mantine-Text-root mantine-1pbxw0k"
          >
            $
            199.6B
          </div>
        </td>
        <td>
          <div
            class="mantine-Text-root mantine-1pbxw0k"
          >
            $
            18.3B
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
